test/Concurrency/sendable_methods.swift

```diff
@@ -0,0 +1,135 @@
+// RUN: %target-typecheck-verify-swift-enable -experimental-feature InferSendableMethods
+// REQUIRES: concurrency
+
+final class C : Sendable {
+  func f() {}
+}
+
+struct S : Sendable {
+  func f(){}
+}
+
+enum E : Sendable {
+  case a, b, c
+  
+  func f() {}
+}
+
+protocol P : Sendable {
+  init()
+}
+
+func g<T>(_ f: @escaping @Sendable (T) -> (@Sendable () -> Void)) where T: P {
+  Task {
+    let instance = T()
+    f(instance)()
+  }
+}
+
+final class InferredSendableC: P {
+  func f() { }
+}
+
+struct InferredSendableS: P {
+  func f() { }
+}
+
+enum InferredSendableE: P {
+  case a, b
+  
+  func f() { }
+}
+
+struct GenericS<T>: P {
+  func f() { }
+}
+
+final class GenericC<T>: P {
+  func f() { }
+}
+
+// Conditional Conformances
+extension GenericS : Sendable where T : Sendable { }
+extension GenericC : Sendable where T : Sendable { }
```

Comment:
`GenericS` and `GenericC` are both always `Sendable` because they unconditionally conform to `P` which refines `Sendable`. I think you can just remove the conformance to `P` for both of these types.